PSModule/AlmLabMgmtClient/SDK/Request/PostRequestBase.cs

```diff
@@ -0,0 +1,45 @@
+using PSModule.AlmLabMgmtClient.SDK.Interface;
+using PSModule.AlmLabMgmtClient.SDK.Util;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PSModule.AlmLabMgmtClient.SDK.Request
+{
+    using C = Constants;
+    public abstract class PostRequestBase : RequestBase
+    {
+        protected PostRequestBase(IClient client) : base(client) { }
+
+        protected override WebHeaderCollection Headers =>
+            new WebHeaderCollection
+            {
+                { HttpRequestHeader.ContentType, C.APP_XML },
+                { HttpRequestHeader.Accept, C.APP_XML },
+                { X_XSRF_TOKEN, _client.XsrfTokenValue }
+            };
+
+        public async override Task<Response> Perform(bool logRequestUrl = true)
+        {
+            return await _client.HttpPost(
+                    Url,
+                    Headers,
+                    GetXmlData(),
+                    ResourceAccessLevel.PROTECTED,
+                    logRequestUrl);
+        }
+
+        private string GetXmlData()
+        {
+            StringBuilder builder = new StringBuilder("<Entity><Fields>");
+            foreach (KeyValuePair<string, string> pair in DataFields)
+            {
+                builder.Append($"<Field Name=\"{pair.Key}\"><Value>{pair.Value}</Value></Field>");

```

Comment:
1. Here `pair.Value` may contains invalid XML characters. Use `CDATA` section to wrap the value.
2. The `pair.Key` may also contains invalid XML characters that may damage entire XML, may need to check before constructing the XML string.

Reply:
1. The `pair.Value` might contain invalid XML characters. For this issue I would ask the QA to reproduce and open a defect, then we can fix and validate it specifically.
What do you think?
2. The `pair.Key` contains always predefined constants with valid characters, at least at this moment.

Reply:
Sure. No problem.